libs/io/bbhnet/io/timeslides.py

```diff
@@ -342,7 +342,33 @@ def shift(self):
 
     def __post_init__(self):
         self.root = Path(self.root)
+        segment = None

```

Comment:
I would either call `self.update()` here, or call `self.__post_init__()` in `self.update`, but calling the same code twice will just make maintaining/updating trickier